cli/src/commands/file/track.rs

```diff
@@ -55,26 +78,220 @@ pub(crate) fn cmd_file_track(
     let matcher = workspace_command
         .parse_file_patterns(ui, &args.paths)?
         .to_matcher();
-    let options = workspace_command.snapshot_options_with_start_tracking_matcher(&matcher)?;
-
-    let mut tx = workspace_command.start_transaction().into_inner();
-    let (mut locked_ws, _wc_commit) = workspace_command.start_working_copy_mutation()?;
-    let (_tree_id, track_stats) = locked_ws.locked_wc().snapshot(&options).block_on()?;
-    let num_rebased = tx.repo_mut().rebase_descendants()?;
-    if num_rebased > 0 {
-        writeln!(ui.status(), "Rebased {num_rebased} descendant commits")?;
+
+    if args.include_ignored {
+        // Bypass ignore rules by manually building the tree from disk
+        let store = workspace_command.repo().store().clone();
+        let working_copy_path = workspace_command.workspace_root().to_path_buf();
+        let auto_tracking_matcher = workspace_command.auto_tracking_matcher(ui)?;
+        let options = workspace_command
+            .snapshot_options_with_start_tracking_matcher(&auto_tracking_matcher)?;
```

Comment:
Should we pass `&matcher` here too? I'm thinking if the user doesn't auto-track any paths and now they run `jj file track --include-ignores untracked ignored` where `untracked` is currently untracked (but not ignored) and `ignore` is untracked and ignored, they will probably want both of them to become tracked. What do you think?